app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeViewModel.kt

```diff
@@ -103,6 +111,42 @@ class HomeViewModel @Inject constructor(
         _showTrendingCoinsLoading.value = false
     }
 
+    @OptIn(FlowPreview::class)
+    fun refresh() = execute {
+        _isRefreshing.value = true
+
+        delay(5000L)
+
+        val getMyCoins = getMyCoinsUseCase.execute(
+            GetMyCoinsUseCase.Input(
+                currency = FIAT_CURRENCY,
+                order = MY_COINS_ORDER,
+                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
+                itemPerPage = MY_COINS_ITEM_PER_PAGE,
+                page = MY_COINS_INITIAL_PAGE
+            )
+        )
+
+        val getTrendingCoins = getTrendingCoinsUseCase.execute(
+            GetTrendingCoinsUseCase.Input(
+                currency = FIAT_CURRENCY,
+                order = MY_COINS_ORDER,
+                priceChangeInHour = MY_COINS_PRICE_CHANGE_IN_HOUR,
+                itemPerPage = MY_COINS_ITEM_PER_PAGE,
+                page = MY_COINS_INITIAL_PAGE
+            )
+        )
```

Comment:
Should we call `getMyCoins` and `getTrendingCoins` instead? By calling that, we can rely on `output.showLoading` in `BaseViewModel` to dismiss the PullToRefresh's spinner (only dismiss the spinner when `showLoading` emits false) 💭

Reply:
> By calling that, we can rely on output.showLoading in BaseViewModel to dismiss the PullToRefresh's spinner

I agree with this, I think we can use just `refresh()` method then we need to handle `showLoading` inside there and then we can remove `getMyCoins` and `getTrendingCoins`

Reply:
@luongvo @Wadeewee There are 2 different loading indicators for MyCoins and TrendingCoins, that's why I separated the refreshing state 😄 

```
    private val _showMyCoinsLoading = MutableStateFlow(false)
    val showMyCoinsLoading: StateFlow<IsLoading>
        get() = _showMyCoinsLoading

    private val _showTrendingCoinsLoading = MutableStateFlow(false)
    val showTrendingCoinsLoading: StateFlow<IsLoading>
        get() = _showTrendingCoinsLoading
```

Reply:
@hoangnguyen92dn I think we control those dialogs by `isRefreshing` Here's some idea:
```
// In HomeScreen

 val showMyCoinsLoading: IsLoading by viewModel.isRefreshing.collectAsState(initial = false)
 val showTrendingCoinsLoading: IsLoading by viewModel.isRefreshing.collectAsState(initial = false)
```

https://user-images.githubusercontent.com/28002315/198924162-cc05ed1e-a5ec-4989-a7ff-6cdb13973adf.mov

Reply:
@Wadeewee I think for initial loading, we are doing right since the UI displays based on completed API responses, therefor we have 2 loading indicators on 2 list then the refreshing indicator doesn't make sense in this scenario. when user pulls down to refresh, it means we reload whole screen data that's why we could keep the indicator until all API completed and refresh the data on screen.

How do you think? @luongvo